libxrdp/xrdp_rdp.c

```diff
@@ -160,9 +161,43 @@ xrdp_rdp_read_config(struct xrdp_client_info *client_info)
                 client_info->use_fast_path = 0;
             }
         }
-        else if (g_strcasecmp(item, "disableSSLv3") == 0)
+        else if (g_strcasecmp(item, "ssl_protocols") == 0)
         {
-            client_info->disableSSLv3 = g_text2bool(value);
+            char s[strlen(value) + 2];
+            snprintf(s, sizeof(s), "%s%s", value, "x");
+
+            /* disable all protocols first, enable later */
+            client_info->ssl_protocols =
+                SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2;
+
+            if (g_pos(value, "TLSv1.2") >= 0)
+            {
+                log_message(LOG_LEVEL_DEBUG, "TLSv1.2 enabled");
+                client_info->ssl_protocols &= ~SSL_OP_NO_TLSv1_2;
+            }
+            if (g_pos(value, "TLSv1.1") >= 0)
+            {
+                log_message(LOG_LEVEL_DEBUG, "TLSv1.1 enabled");
+                client_info->ssl_protocols &= ~SSL_OP_NO_TLSv1_1;
+            }
+            if (g_pos(value, "TLSv1 ") >= 0 || g_pos(s, "TLSv1x") >= 0)
```

Comment:
Actually, it's not that bad. But let's add " " instead of "x". Then we can match all strings with the final space: "SSLv3 ", "TLSv1 ", "TLSv1.1 " and so on.